PrivacyIDEA_Client/PrivacyIDEA.cs

```diff
@@ -286,35 +293,36 @@ public PIResponse ValidateCheck(string user, string otp, string? transactionid =
             headers ??= new List<KeyValuePair<string, string>>();
             headers.Add(new KeyValuePair<string, string>("Origin", origin));
 
-            string response = SendRequest("/validate/check", parameters, headers);
+            string response = await SendRequest("/validate/check", parameters, "POST", headers, cancellationToken);
             return PIResponse.FromJSON(response, this);
         }
 
         /// <summary>
         /// Gets an auth token from the privacyIDEA server using the service account.
         /// Afterward, the token is set as the default authentication header for the HttpClient.
         /// </summary>
+        /// <param name="cancellationToken">optional</param>
         /// <returns>true if success, false otherwise</returns>
-        private bool GetAuthToken()
+        private async Task<bool> GetAuthToken(CancellationToken cancellationToken = default)
         {
             if (!ServiceAccountAvailable())
             {
                 Error("Unable to fetch auth token without service account!");
                 return false;
             }
 
-                var map = new Dictionary<string, string>
-                    {
-                        { "username", serviceuser },
-                        { "password", servicepass }
-                    };
+            var parameters = new Dictionary<string, string>
+                {
+                    { "username", _Serviceuser }, // ServiceAccountAvailable already passed
```

Comment:
i do not think the comment is necessary here

Reply:
Is removed in #4